### PR TITLE
PA-4752 Refine DAST Authentication

### PR DESCRIPTION
**Pull request by SOOS-JAlvarez**
### Changes:
- Possibility to look for tokens in newly setup cookies/sessionStorage/localStorage
- Changed to ChromeDriver instead of using Firefox One
- Added param bearerToken to add to every request

**Ticket:** https://soos.atlassian.net/browse/PA-4752

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -690,6 +719,13 @@ def parse_args(self) -> None:
             help="First submit button id to use in auth apps",
             required=False,
         )
+        parser.add_argument(
+            "--authSubmitAction",
+            help="Submit action to perform on form filled, click or submit",
+            type=str,
+            nargs="*",
```
Comment (SOOS-MMalony) [inline main.py:726]: Should this be nargs? I think that's going to allow the user to specify multiple times & give you an array value.

File: main.py [hunk 2]
```diff
@@ -799,7 +843,8 @@ def run_analysis(self) -> None:
             log(f"Target URL: {self.target_url}")
             print_line_separator()
 
-            check: bool = check_site_is_available(self.target_url)
+            check = True
+            # bool = check_site_is_available(self.target_url)
```
Comment (SOOS-MMalony) [inline main.py:847]: Did you mean to leave this commented out? If we don't need the check we should just pull these lines out.

File: helpers/auth.py [hunk 3]
```diff
@@ -216,14 +226,15 @@ def login(self):
...
             element = self.find_element(
                 submit_field_name, "submit", "//*[@type='submit' or @type='button' or button]")
-            element.click()
+            actions = ActionChains(self.driver)
+            actions.move_to_element(element).click().perform()
             log(f"Clicked the {submit_field_name} element")
-        elif username_element:
-            username_element.submit()
+        elif password_field_name:
+            self.find_element(password_field_name,"password","//input[@type='password' or contains(@name,'ass')]",)
```
[the remark this answers is not in the document]
Comment (SOOS-JAlvarez) [reply helpers/auth.py]: yep, forgot to add the .submit() at the end, I've just added it.

File: main.py (hunk 2 above)
Comment (SOOS-JAlvarez) [reply main.py]: Comented while testing to not save a bit of time on each run. back to normal.

**Review by SOOS-MMalony — APPROVED** [review]

File: main.py [hunk 4]
```diff
@@ -300,8 +300,8 @@ def __add_zap_options__(self, args: List[str]) -> NoReturn:
         if self.auth_password_field_name is not None:
             zap_options.append(self.__add_custom_option__(label="auth.password_field", value=self.auth_password_field_name))
         
-        # ZAP options should be wrapped with ""
-        if len(zap_options) > 0:
+        # ZAP options should be wrapped with "" when automatic auth is enabled
+        if len(zap_options) > 0 and self.auth_loginUrl is not None:
```
Comment (SOOS-JAlvarez) [inline main.py:304]: just to have it somewhere when passing --requestHeader it's necessary to wrap everything with single quotes.
eg: `--requestHeader="'authorization:Bearer tokenValue'"`

**Review by SOOS-MMalony — APPROVED** [review]

**Pull request merged**